### ISSUE TITLE
[Bug] Got stucked when running python script from a shell script 

### ISSUE THREAD

**Issue by ilongshan** [ray-project/ray#21298]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Core

### What happened + What you expected to happen

With the following shell script, the program got stucted. But it is ok with the python script.

### Versions / Dependencies

OS: macOS Big Sur Apple M1
Python: 3.9.7
Ray: 1.9.1
ffmpeg: 4.4.1

### Reproduction script

1. Python script: filename=test_ray.py
```Python, name=test_ray.py
#!/usr/bin/env python3
# -*- coding:utf-8 _*-

import ray
import subprocess
import shlex

ray.init(num_cpus=4)


@ray.remote
def count(i):
    decoder = f"ffmpeg -y " \
              f"-i enc.h265 " \
              f"-c:v rawvideo " \
              f"-pix_fmt yuv420p " \
              f"{str(i)}_dec.yuv"
    decoder_args = shlex.split(decoder)
    dec_log = "test_dec.log"
    with open(dec_log, "w") as dec_output:
        decoder_result = subprocess.run(decoder_args, stdout=dec_output,
                                        stderr=subprocess.STDOUT)
        print(decoder_result)


if __name__ == "__main__":
    count_vec = [count.remote(i) for i in range(10)]
    ray.get(count_vec)


```

2. Shell script: filename=test_ray.sh
```shell, name=test_ray.sh
#!/usr/bin/env bash
#-*- coding:utf-8 _*-

export PYTHONPATH=$PYTHONPATH:.

python test_ray.py
```

3. Download and extract the attched "enc.h265.zip" under the same directory with test_ray.sh and test_ray.py.
4.  Run the shell script
```shell
bash test_ray.sh
```
4. Log on terminal
```
2021-12-30 18:10:39,160 INFO services.py:1338 -- View the Ray dashboard at http://127.0.0.1:8265
```
5. We can see it is stucked 
[enc.h265.zip](https://github.com/ray-project/ray/files/7792479/enc.h265.zip)

### Anything else

It can be produced when running with the shell script, but is ok when running the python script.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by scv119** [comment]
cc @clarkzinzow maybe worth a quick reproduction? not sure if it's M1 specific issue.

**Comment by scv119** [comment]
hmm seems I can reproduce this issue with the above patch.

**Comment by scv119** [comment]
I wonder if there is anything specific to ffmpeg. I changed the code from
```
    decoder = f"ffmpeg -y " \
              f"-i enc.h265 " \
              f"-c:v rawvideo " \
              f"-pix_fmt yuv420p " \
              f"{str(i)}_dec.yuv"
```
to 
```
    decoder = "sleep 1"
```

then anything seems fine.

**Comment by ilongshan** [comment]
> hey ilongshan, seems your script has a bug where you didn't call `ray.get(count_vec)` after
> 
> ```
> if __name__ == "__main__":
>     count_vec = [count.remote(i) for i in range(10)]
> ```

Accidentally delete this line when clean the code. Thanks for modify the code.

**Comment by ilongshan** [comment]
> I wonder if there is anything specific to ffmpeg. I changed the code from
> 
> ```
>     decoder = f"ffmpeg -y " \
>               f"-i enc.h265 " \
>               f"-c:v rawvideo " \
>               f"-pix_fmt yuv420p " \
>               f"{str(i)}_dec.yuv"
> ```
> 
> to
> 
> ```
>     decoder = "sleep 1"
> ```
> 
> then anything seems fine.

Yes it seems relate to ffmpeg. But ffmpeg is just a commonly used command line tools. And I already launch many ffmpeg simulanously. And these ffmpeg processes work fine.

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
